internal/services/v1/permissions.go

```diff
@@ -427,16 +430,47 @@ func (ps *permissionServer) LookupSubjects(req *v1.LookupSubjectsRequest, resp v
 	usagemetrics.SetInContext(ctx, respMetadata)
 
 	stream := dispatchpkg.NewHandlingDispatchStream(ctx, func(result *dispatch.DispatchLookupSubjectsResponse) error {
-		for _, foundSubject := range result.FoundSubjects {
+		foundSubjects, ok := result.FoundSubjectsByResourceId[req.Resource.ObjectId]
+		if !ok {
+			return fmt.Errorf("missing resource ID in returned LS")
+		}
+
+		for _, foundSubject := range foundSubjects.FoundSubjects {
 			excludedSubjectIDs := make([]string, 0, len(foundSubject.ExcludedSubjects))
 			for _, excludedSubject := range foundSubject.ExcludedSubjects {
 				excludedSubjectIDs = append(excludedSubjectIDs, excludedSubject.SubjectId)
 			}
```

Comment:
We err on the side of safety by keeping the deprecated excluded subject IDS with a conservative value that considers subjects for which caveat evaluation missed context as excluded.